### ISSUE TITLE
HTTPS

### ISSUE THREAD

**Issue by metalralf** [detarkende/stremio-ncore-addon#17]
Based on the other guy's nCore addon from your reddit post, I think it's fairly easy to make https happen.
https://github.com/nyakaspeter/stremio-torrent-stream/blob/main/src/utils/https.ts

**Comment by x0r3n** [comment]
Easy to use with reverse proxy, like nginx and you can add letsencrypt for public cert. And you can use then addon from every where also if you put it to public

**Comment by detarkende** [comment]
@metalralf the current recommended solution is to put the addon behind a reverse proxy (either directly with nginx or something similar, or with cloudflare tunnels).

What's your recommended alternative to this that keeps the current solutions possible?

**Comment by slashwarm** [comment]
I use a modified version of the source code, that sets up the node server with https://local-ip.medicmobile.org/'s SSL certificates. This way I can reach the addon through `*.local-ip.medicmobile.org:3000`. Now this only works on local network, but that is fine for me. The reason I'm not using http://local-ip.co/ is because their certificate recently expired and I was too impatient to wait for new ones. Also, letting the certificate expire and not renew it for weeks doesn't seem too reliable.

```
serve({
  fetch: baseApp.fetch,
  port: config.PORT,
  createServer: createSecureServer,
  serverOptions: {
    key: readFileSync('/etc/ssl/private/server.key'),
    cert: readFileSync('/etc/ssl/private/server.pem'),
    ca: readFileSync('/etc/ssl/private/chain.pem'),
  },
});
```

Previously I used localtunnel however it turned out to be a piece of shit, I would not waste my time implementing it. Every second request would return a HTTP 500 and even when it worked, it was slow and buggy as hell. Although it is much slower than using local-ip, but if you want public access to the addon, I would recommend `https://telebit.cloud/ `. It is a free tunnel that provides a fixed domain, unlike Ngrok which doesn't provide a fixed domain in its free tier. Or you could just use Cloudflare's tunnel as the setup guide suggests. I'm not sure whether they provide free domains, I heard they do. Dunno, never used it. Here is a cool list of tunneling services: https://github.com/anderspitman/awesome-tunneling. Feel free to test them if you have too much free time on your hands.

In conclusion, I suggest adding support for [local-ip.medicmobile.org](https://local-ip.medicmobile.org/) as I have found it to be the fastest and most reliable solution. Other tunneling services, that open ports to the internet don't require any special support from the addon.

**Comment by nyakaspeter** [comment]
Hi all, I'm "the other guy" 😄 I agree with @slashwarm, I don't recommend any tunneling. Use local-ip.medicmobile on LAN (hope it won't expire like local-ip) or Tailscale if you want to access the addon from anywhere, it works fine, free and pretty easy to set up. Or use a reverse proxy if you have a domain.
